core/src/impl/Kokkos_Profiling.cpp

```diff
@@ -97,13 +108,31 @@ bool eventSetsEqual(const EventSet& l, const EventSet& r) {
          l.end_fence == r.end_fence && l.sync_dual_view == r.sync_dual_view &&
          l.modify_dual_view == r.modify_dual_view &&
          l.declare_metadata == r.declare_metadata &&
+         l.request_settings == r.request_settings &&
+         l.transmit_actions == r.transmit_actions &&
          l.declare_input_type == r.declare_input_type &&
          l.declare_output_type == r.declare_output_type &&
          l.end_tuning_context == r.end_tuning_context &&
          l.begin_tuning_context == r.begin_tuning_context &&
          l.request_output_values == r.request_output_values &&
          l.declare_optimization_goal == r.declare_optimization_goal;
 }
+template <typename Callback, typename BooleanConstant, typename... Args>
+inline void invoke_kokkos_callback(const Callback& callback, BooleanConstant,
+                                   Args&&... args) {
+  if (callback != nullptr) {
+    // two clause if statement
+    // BooleanConstant::value: "if this callback ever needs a fence", AND
+    // if the tool requires global fencing (default true, but tools can
+    // overwrite)
+    if ((BooleanConstant::value) &&
```

Comment:
Instead of just a comment, how about replacing `BooleanConstant` (which is a description of the type and not its purpose) with something like `enum class RequiresGlobalFencing : bool { Yes, No } `. Given that this is inlined, the check will likely always be optimized away (or use a two different functions differentiated by a tag or use RequiresGlobalFencing as a non-type template parameter that we specify explicitly if we are really concerned about this).

I'm also wondering if `Callback` and `RequiresGlobalFencing` should be switched, as `callback` and its `args` should be together.